samcli/lib/build/build_strategy.py

```diff
@@ -0,0 +1,240 @@
+"""
+Keeps implementation of different build strategies
+"""
+import logging
+import pathlib
+import shutil
+
+from samcli.commands.build.exceptions import MissingBuildMethodException
+from samcli.lib.utils import osutils
+from samcli.lib.utils.hash import dir_checksum
+
+LOG = logging.getLogger(__name__)
+
+
+class BuildStrategy:
+    """
+    Base class for BuildStrategy
+    Keeps basic implementation of build, build_functions and build_layers
+    """
+
+    def __init__(self, build_graph):
+        self._build_graph = build_graph
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
```

Comment:
What's `exc_tb`? Is there a more descriptive name?

Reply:
Tangentially: why are we implementing the class as a context manager if `__enter__` and `__exit__` are no-ops?

Reply:
`__exit__` method is used in cached build implementation, that is why we need context.

Thanks for suggestion @hoffa , it would be better to replace `__enter__` and `__exit__` methods with extending `AbstractContextManager` class.

Reply:
Synced offline with @hoffa, and decided to keep as it is.